services/command-router-base/src/main/java/org/eclipse/hono/commandrouter/impl/CommandRouterServiceImpl.java

```diff
@@ -209,53 +220,94 @@ private Duration getSanitizedLifespan(final Duration lifespan) {
     public Future<CommandRouterResult> enableCommandRouting(final List<String> tenantIds, final Span span) {
         Objects.requireNonNull(tenantIds);
         final boolean isProcessingRequired = tenantsToEnable.isEmpty();
-        tenantsToEnable.addAll(tenantIds);
+        tenantIds.forEach(id -> {
+            tenantsToEnable.addLast(Pair.of(id, 1));
```

Comment:
How about checking for each to-be-added tenantId first, if it's already in `tenantsToEnable`?
Otherwise the `attemptNo` counter wouldn't be accurate.